oteapi/utils/paths.py

```diff
@@ -31,13 +31,10 @@ def uri_to_path(uri: "Union[str, AnyUrl, ParseResult]") -> Path:
         A properly converted URI/IRI/URL to `pathlib.Path`.
 
     """
-    if isinstance(uri, (AnyUrl, ParseResult)):
-        uri_path = uri.path
-    elif isinstance(uri, str):
-        uri = urlparse(uri)
-        uri_path = uri.path
-    else:
-        raise TypeError("uri is expected to be either a string or parsed URI/IRI/URL.")
+    if not isinstance(uri, ParseResult):
+        uri = urlparse(uri)  # type: ignore
+
+    uri_path = uri.netloc + uri.path if uri.scheme == "file" else uri.path
```

Comment:
To avoid precedence problems (or confusing readability), please use parentheses to clarify it the `if` statement should include the expression `uri.netloc + uri.path` or if you are just checking the `uri.path`

Reply:
done